### ISSUE TITLE
[Dialog] No longer computes and assigns a z-index

### ISSUE THREAD

**Issue by colepeters** [radix-ui/primitives#1086]
### Current Behavior

Since 0.1.5, the Dialog primitive no longer computes and assigns a z-index to the component (as it did in 0.1.1).

### Expected behavior

The maximum z-index should be calculated and assigned to the component, as it was before. Or, if this is an intentional change, documentation should be updated to account for this.

### Reproducible example

[See this demo on CodeSandbox](https://codesandbox.io/s/admiring-grass-ws06k?file=/src/App.js)

When opening the Dialog, the H1 element will show as overlaying the Dialog's content.

### Your environment

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Radix Package(s) |  react-dialog       |   0.1.5      |
| React            |      |   17.x      |
| Browser          |   Chrome      |     latest    |
| Assistive tech   |   n/a      |         |
| Node             | n/a     |         |
| npm/yarn         |   npm      |    7.20     |
| Operating System |   macOS      |    11.6     |

**Comment by benoitgrelard** [comment]
Hi @colepeters,

It is a change and the documentation has been updated in the release notes as well as each component pages.
This is because we now expose a new `Dialog.Portal` part so you have full access to the portal and how you want to position/layer things. Which means you now have full access to set your own z-index (if required).

This not only enables uses to fix layering issues they might have had when using these primitives in conjunction with other libs, but also enables some use-cases that weren't possible before like nesting the content inside the overlay to have an outer-scroll.

I'll close the issue for now, but if that isn't satisfactory, feel free to keep discussing here.

👍

**Comment by colepeters** [comment]
Thanks for the info, @benoitgrelard. I suppose the release notes and the component documentation weren't super clear to me that I'd now have to set my own z-index, but maybe that's because I'm coming from a previous version where this was managed by the component. I did see the note about the new Portal component (which I used in my sandbox), but the fact of that component being exposed didn't really make the think that I'd now need to handle the z-index myself. (To be clear, I'm not complaining about having to handle it, this was just unexpected.)

Maybe this is just a case of being biased by a previous version. :)
